src/allotropy/parsers/thermo_fisher_genesys30/thermo_fisher_genesys30_structure.py

```diff
@@ -0,0 +1,147 @@
+""" Structure file for ThermoFisher Genesys 30 Adapter """
+from __future__ import annotations
+
+from datetime import datetime
+import io
+
+import pandas as pd
+
+from allotropy.allotrope.models.adm.spectrophotometry.benchling._2023._12.spectrophotometry import (
+    ContainerType,
+)
+from allotropy.allotrope.models.shared.definitions.definitions import (
+    FieldComponentDatatype,
+)
+from allotropy.allotrope.schema_mappers.adm.spectrophotometry.benchling._2023._12.spectrophotometry import (
+    Data,
+    DataCube,
+    DataCubeComponent,
+    Measurement,
+    MeasurementGroup,
+    MeasurementType,
+    Metadata,
+)
+from allotropy.exceptions import AllotropeConversionError
+from allotropy.named_file_contents import NamedFileContents
+from allotropy.parsers.constants import NOT_APPLICABLE
+from allotropy.parsers.lines_reader import LinesReader, read_to_lines
+from allotropy.parsers.thermo_fisher_genesys30 import constants
+from allotropy.parsers.utils.pandas import df_to_series_data
+from allotropy.parsers.utils.uuids import random_uuid_str
+from allotropy.parsers.utils.values import try_float_or_none
+
+
+def create_data(named_file_contents: NamedFileContents) -> Data:
+    """
+    Structures the named_file_contents according to the mapper file
+    :param named_file_contents: The contents of the file to convert
+    :return: The data according to spectrophotometer data mapper
+    """
+    filename = named_file_contents.original_file_name
+
+    lines = read_to_lines(named_file_contents)
+    reader = LinesReader(lines)
+
+    if filename.endswith(".csv"):
+        metadata_list = [line for line in reader.pop_until("^,,") if line]
+        reader.drop_until_inclusive(",,")
+        delimiter = ","
+    elif filename.endswith(".tsv"):
+        metadata_list = list(reader.pop_until_empty())
+        reader.drop_until_inclusive("")
+        delimiter = "\t"
+    else:
+        message = f"{constants.UNSUPPORTED_FILE_FORMAT_ERROR} {filename}"
+        raise AllotropeConversionError(message)
+
+    rawdata_list = list(reader.pop_until_empty())
+    rawdata_string = io.StringIO("\n".join(rawdata_list))
+    rawdata_dataframe = pd.read_csv(rawdata_string, header=0, delimiter=delimiter)
+
+    metadata_string = io.StringIO("\n".join(metadata_list))
+    metadata_dataframe = pd.read_csv(
+        metadata_string,
+        header=None,
+        delimiter=delimiter,
+        keep_default_na=False,
+        index_col=0,
+    ).T
+    metadata_dataframe.columns = metadata_dataframe.columns.str.strip()
```

Comment:
Please call `as_type(str)` on the dataframe before calling `.str.strip()` we have seen errors otherwise when a value is interpreted as a non-string.

You can call this when creating the dataframe above.

It is OK to cast as a string, as all `SeriesData` accessors start from a string anyway.

Reply:
Okay, Nate, we'll cast it to a string to avoid any errors due to type differences. Thanks for catching this!